File: application/src/main/java/de/wuespace/telestion/project/daedalus2/TimeSyncRequest.java
package de.wuespace.telestion.project.daedalus2;

import com.fasterxml.jackson.annotation.JsonProperty;
import de.wuespace.telestion.api.message.JsonMessage;

@SuppressWarnings("unused")
public record TimeSyncRequest(
		@JsonProperty String target,
		@JsonProperty String command
) implements JsonMessage {
	public TimeSyncRequest() {
		this(null, null);
	}
}

File: client/src/widgets/time-sync-widget/time-sync-request.ts
export interface TimeSyncRequest {
	className: 'de.wuespace.telestion.project.daedalus2.timesync.TimeSyncRequest';
	target: 'ejector' | 'seedA' | 'seedB';
	command: 'time setfw' | 'time force' | 'time forcefw';
}
